src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspace.cs

```diff
@@ -32,44 +34,57 @@ static VisualStudioWorkspace()
         internal VisualStudioWorkspace(HostServices hostServices)
             : base(hostServices, WorkspaceKind.Host)
         {
-            _backgroundCompiler = new BackgroundCompiler(this);
-
-            var cacheService = Services.GetService<IWorkspaceCacheService>();
-            if (cacheService != null)
+            _backgroundCompilationDisabled = this.Services.GetRequiredService<IWorkspaceConfigurationService>().Options.DisableBackgroundCompilation;
```

Comment:
❔ Are feature flags updated by this point?

Reply:
It's all done async by the platform. So we will read the value as is at startup. Then, when it changes we will get the fresh value on next session.